concent_api/core/tests/test_core_views.py

```diff
@@ -286,6 +286,70 @@ def test_send_reject_message_save_as_receive_out_of_band_status(self):
         self.assertEqual(reject_response.status_code, 202)
         self.assertEqual(Message.objects.last().type, message.RejectReportComputedTask.TYPE)
 
+    @freeze_time("2017-11-17 10:00:00")
+    def test_send_should_return_http_400_if_task_to_compute_deadline_is_negative_integer(self):
+        compute_task_def = message.ComputeTaskDef()
+        compute_task_def['task_id'] = 8
+        compute_task_def['deadline'] = -11
+        task_to_compute = message.TaskToCompute(
+            timestamp = self.message_timestamp,
+            compute_task_def = self.compute_task_def,
+        )
+
+        serialized_task_to_compute      = dump(task_to_compute,             REQUESTOR_PRIVATE_KEY,   PROVIDER_PUBLIC_KEY)
+        deserialized_task_to_compute    = load(serialized_task_to_compute,  PROVIDER_PRIVATE_KEY,  REQUESTOR_PUBLIC_KEY, check_time = False)
+
+        ack_report_computed_task = message.AckReportComputedTask(
+            timestamp = self.message_timestamp,
+        )
+        ack_report_computed_task.task_to_compute = deserialized_task_to_compute
+
+        response_400 = self.client.post(
+            reverse('core:send'),
+            data = dump(
+                ack_report_computed_task,
+                PROVIDER_PRIVATE_KEY,
+                CONCENT_PUBLIC_KEY
+            ),
+            content_type                    = 'application/octet-stream',
+            HTTP_CONCENT_CLIENT_PUBLIC_KEY  = b64encode(PROVIDER_PUBLIC_KEY).decode('ascii'),
+        )
+
+        self.assertEqual(response_400.status_code, 400)
+        self.assertIn('error', response_400.json().keys())
+
+    @freeze_time("2017-11-17 10:00:00")
+    def test_send_should_return_http_400_if_task_to_compute_deadline_cant_be_casted_to_int(self):
+        compute_task_def = message.ComputeTaskDef()
+        compute_task_def['task_id'] = 8
+        compute_task_def['deadline'] = 'a11b'
```

Comment:
You're missing a test with the timestamp in the future.

Actually, these tests are almost identical. Maybe replace them with one that has lists of valid and invalid values and checks them in a loop. We'll be able to test more values with less code this way.

Reply:
:+1:

Reply:
Looks like I got this one confused with #61. What I meant was to add more invalid values. Like dicts, objects, `None`, etc. I remember that I listed them in a comment in this pull request but I can't find that comment anywhere.